### ISSUE TITLE
TF 2.0 Beta MobileNetV2 model.fit() and model.evaluate() works unexpectedly 

### ISSUE THREAD

**Issue by yichenj** [tensorflow/tensorflow#31473]
Dear Expert,
I am using TF 2.0-beta to have some tests based on the tf.keras API, and I think the result of MobileNetV2 model.fit(evaluate) is not reasonable. I doubt it is a bug, and I currently only found it in MobileNetV2 model. Could you please help to have a look? The details follows below.

Thanks in advance.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04

- TensorFlow installed from (source or binary): pip

- TensorFlow version (use command below): 2.0.0-beta

- Python version: 3.5.2

- CUDA/cuDNN version: 10.0
- GPU model and memory: Tesla P4

**Describe the current behavior**
Training a MobileNetV2 model , I see the loss is descending and becomes very small after a few epochs while the val_loss is not.  During my checking progress, I intentionally use the training data to evaluate the model, and i can see the evaluation result is quite different from the training progress shows.

The program output is attached as 1.log.
[1.log](https://github.com/tensorflow/tensorflow/files/3486313/1.log)

**Describe the expected behavior**
I do not believe such a huge gap comes from BN/Dropout or the batch difference, I doubt there is a bug in computing the loss in keras.model.fit()? 
Besides, I print the prediction result array. The value changes from different inputs are very small, it seems to me this is under-fitting? However, from the loss curve I cannot deduce it..(More like a over-fitting case to me..). Could you please help to figure it out?

**Code to reproduce the issue**

    # !!! I loaded the model as below:
    model = keras.applications.mobilenet_v2.MobileNetV2(input_shape=(299, 299, 3), include_top=True, weights=None, classes=CATEGORY_SIZE)
    optimizer = tf.keras.optimizers.Adam(learning_rate=0.001)
    model.compile(optimizer=optimizer,
                  loss='sparse_categorical_crossentropy',
                  metrics=['accuracy'])
    model.summary()

    print('###Trainin begin...###')
    # !!! I loaded the model as below:
    model = keras.applications.mobilenet_v2.MobileNetV2(input_shape=(299, 299, 3), include_top=True, weights=None, classes=CATEGORY_SIZE)
    optimizer = tf.keras.optimizers.Adam(learning_rate=0.001)
    model.compile(optimizer=optimizer,
                  loss='sparse_categorical_crossentropy',
                  metrics=['accuracy'])
    model.summary()
    model.fit(train_generator,
              epochs=200,
              callbacks=[],
              shuffle=True,
              validation_data = test_generator)

    # !!! I intentionally use the training data to evaluate the model, 
    # !!! expect that the loss and val should be almost equal to the value in fit() progess?
    print('###Evaluate with training data###')
    model.evaluate(train_generator)
    print('###Evaluate with testing data###')
    model.evaluate(test_generator)

    # !!! Use an array instead of a generator and print the prediction array see what happens
    # !!! All the prediction arrays gets the similar value from different inputs. Is this under-fitting?
    print('###Predict images###')
    i = 0
    test_images = np.empty((0, 299, 299, 3))
    test_labels = np.empty(0)
    for x, y in test_generator:
        i = i + 1
        if i > len(test_generator):
            break
        test_images = np.concatenate((test_images, x), axis=0)
        test_labels = np.concatenate((test_labels, y), axis=0)

    error_count = 0
    predictions = model.predict(test_images)
    for i in range(test_images.shape[0]):
        predicted_label = np.argmax(predictions[i])
        if not predicted_label == test_labels[i]:
            error_count = error_count + 1
            print("Label for image %d: %d" % (i, test_labels[i]))
            print("Predict label for image %d: %d" % (i, predicted_label))
            print(np.around(predictions[i], decimals=3))
        print("Total error count is %d, total predict count is %d" % (error_count, test_images.shape[0]))

**Comment by yichenj** [comment]
Not an issue.. My mistake for the inputs

**Comment by yichenj** [comment]
Seems that the model has got over-fitting. The most recent 3/4 batches always got accurate result but the others got poor fitting.
